test/metric/procstat.go

```diff
@@ -21,6 +21,7 @@ var _ MetricValueFetcher = (*ProcStatMetricValueFetcher)(nil)
 
 func (f *ProcStatMetricValueFetcher) Fetch(namespace, metricName string, stat Statistics) (MetricValues, error) {
 	dims := f.getMetricSpecificDimensions()
+	dims = append(dims, f.getInstanceIdDimension())
```

Comment:
Do we always want to pull in metric dimensions as well as instance dimensions? If so, should we just go back to the interface and consolidate there so we don't add `append` everywhere?

So `getDimensions()` instead of `getMetricSpecificDimensions()` + `getInstanceIdDimensions()`?

Reply:
We don't. Well this got moved in here because of ECS test. And now I'm actually realizing some configurations don't have instance id as dimension. (If append-dimension isn't specified). I'm in process of adding a simple test for behavior without append-dimension, and will have some modifications for this dimension logic while doing that. Mind if we punt this convo til my test PR?